research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,233 @@
+
+### Screen reader accessibility
+
+> Review/link aria specifications and list which aria labels should appear on which parts of the component.
+>
+> If there are ambiguous requirements or lacking guidance in the specifications, consult the experts in your organization to propose refinements/additions to the specifications.
+>
+> Example:
+>
+> `root`:
+>
+> - should render the native element using the `as` prop, defaulting to `div`
+> - should mix in native props expected for the element type defined in `as`.
+>
+> `thumb`:
+>
+> - should be focusable via `tabindex=0`
+> - role set to `{ComponentName}`
+> - receives `aria-valuemin` and `aria-valuemax` representing min/max
+> - receives `aria-valuenow` representing the current value
+
+### Accessibility concerns for the user
+
+> Consider which props, mapped to attribute values, must be provided for the user to make the component accessible. Often props which require localization or calculating a value must be provided by the user.
+>
+> For React props, consider using `ariaLabel` and `ariaLabelledBy`, rather than the natural kebab-cased variants (`aria-label`) so that they can be provided without destroying type safety.
+
+## Themability and customization
+
+> Components should be built to withstand multiple aesthetic looks. This ensures customizability and reuse in the long term.
+>
+> At minimum, each component should consider the following theming requirements: light mode, dark mode, high contrast light and high contrast dark.
+>
+> If applicable, include some screenshots of themed variants of the component to get an idea of what is to be achieved through customization.
+
+### Class names
+
+> Components should include a class name contract to achieve maximum reusability. The contract should include:
+>
+> - A classname per anatomy (Examples: `root`, `rail`, `thumb`)
+> - A classname per state/variant attached to the root in applicable conditions (Examples: `rootDisabled`, `rootSelected`)
+>
+> Example:
+>
+> | Name           | Description                             |
+> | -------------- | --------------------------------------- |
+> | `root`         | Root slot class.                        |
+> | `rail`         | Rail slot class.                        |
+> | `thumb`        | Thumb slot class.                       |
+> | `rootDisabled` | Applied when the component is disabled. |
+> | `rootSelected` | Applied when the component is selected. |
+
+### Component design tokens
+
+> Assuming the Fluent UI styling system, a component's styling may reference design tokens.
+>
+> Tokens represent the general look and feel of the various visual slots. Tokens feed values into the styling during applicable conditions. For example, `railBackgroundHovered` would apply to the `rail` background when the mouse cursor is hovered over the component.
```

Comment:
Since this is being set as a template, does it make sense to start a conversation on terminology? For instance, what do we mean when we say `slot` - are we following the Web Component pattern here or is it slightly different? May not be the time, but figured I would float it.

Reply:
A `slot` is a named part of a component representing a placeholder location to replace with content, but usually having a default element type. It is typically represented by a single DOM element.

Reply:
Should we add terminology in the template? Or somewhere else?

I could certainly take a stab at it. I could open a bug, and get this checked in and follow up in a separate pr

Reply:
I'm good with either - I don't think it needs to block this, but we should definitely have the conversation as I think semantics will play a big part of what we're trying to accomplish here.